stages/org.osbuild.machine-id

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+"""
+Deal with /etc/machine-id
+
+Explicitly define the state to /etc/machine-id.
+"""
+
+import pathlib
+import sys
+
+import osbuild.api
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["machine-id"],
+"properties": {
+  "machine-id": {
+    "enum": ["first-boot", "empty", "preserve"],
+    "description": "Ensure the state of the /etc/machine-id file in the tree"
+  }
+}
+"""
+
+
+def main(tree, options):
+    mode = options["machine-id"]
+
+    # use "match" here once we are on py3.10
+    machine_id_file = pathlib.Path(f"{tree}/etc/machine-id")
+    if mode == "first-boot":
+        # available since systemd v247, systemd PR#16939
+        machine_id_file.write_bytes(b"uninitialized\n")
+    elif mode == "empty":
+        with open(machine_id_file, "wb") as fp:
+            fp.truncate(0)
+    elif mode == "preserve":
+        if not machine_id_file.is_file():
+            print(f"{tree}/etc/machine-id cannot be preserved, it does not exist")
+            return 1
```

Comment:
We've spoken about this before in regards to where we depend on the schema and where we don't. Let's add an `else` here to raise a `ValueError` just to keep locality.